File: app/src/main/java/org/intelehealth/app/activities/visitSummaryActivity/BluetoothDeviceAdapter.java
package org.intelehealth.app.activities.visitSummaryActivity;

import android.bluetooth.BluetoothDevice;
import android.content.Context;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import org.intelehealth.app.R;

import java.util.List;

public class BluetoothDeviceAdapter extends BaseAdapter {

    private Context mContext;
    private List<BluetoothDevice> mList;
    private LayoutInflater mInflater;

    public BluetoothDeviceAdapter(Context context, List<BluetoothDevice> list) {
        this.mContext = context;
        this.mList = list;
        mInflater = LayoutInflater.from(mContext);
    }

    @Override
    public int getCount() {
        return mList.size();
    }

    @Override
    public Object getItem(int position) {
        return mList.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    private class ViewHolder{
        TextView tvName;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        ViewHolder holder;
        if(convertView == null){
            convertView = mInflater.inflate(R.layout.bluetooth_device_item,null);
            holder = new ViewHolder();
            holder.tvName = (TextView) convertView.findViewById(R.id.tv_bluetooth_device_name);
            convertView.setTag(holder);
        }else {
            holder = (ViewHolder) convertView.getTag();
        }
        BluetoothDevice bluetoothDevice = mList.get(position);
        if(TextUtils.isEmpty(bluetoothDevice.getName())){
            holder.tvName.setText(bluetoothDevice.getAddress());
        }else {
            holder.tvName.setText(mList.get(position).getName()  + " [" + bluetoothDevice.getAddress() + "]");
        }
        return convertView;
    }
}

File: app/src/main/java/org/intelehealth/app/activities/prescription/thermalprinter/BluetoothDeviceAdapter.kt
package org.intelehealth.app.activities.prescription.thermalprinter

import android.bluetooth.BluetoothDevice
import android.content.Context
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import org.intelehealth.app.R

class BluetoothDeviceAdapter(private val mContext: Context, private val mList: List<BluetoothDevice>) : BaseAdapter() {

    private val mInflater: LayoutInflater = LayoutInflater.from(mContext)

    override fun getCount(): Int {
        return mList.size
    }

    override fun getItem(position: Int): Any {
        return mList[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    private class ViewHolder {
        var tvName: TextView? = null
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val holder: ViewHolder
        var view = convertView

        if (view == null) {
            view = mInflater.inflate(R.layout.item_bluetooth_devices, parent, false)
            holder = ViewHolder()
            holder.tvName = view.findViewById(R.id.tv_bluetooth_device_name)
            view.tag = holder
        } else {
            holder = view.tag as ViewHolder
        }

        val bluetoothDevice = mList[position]
        holder.tvName?.text = if (TextUtils.isEmpty(bluetoothDevice.name)) {
            bluetoothDevice.address
        } else {
            "${bluetoothDevice.name} [${bluetoothDevice.address}]"
        }

        return view!!
    }
}